crud/stats/metrics_registry.lua

```diff
@@ -0,0 +1,323 @@
+--  age buckets. `metrics >= 0.5.0, < 0.9.0` is unsupported
+--  due to quantile overflow bug
+--  (https://github.com/tarantool/metrics/issues/235).
+--
+-- @function is_supported
+--
+-- @treturn boolean Returns true if `metrics >= 0.9.0` found, false otherwise.
+--
+function registry.is_supported()
+    if is_package == false then
+        return false
+    end
+
+    -- Only metrics >= 0.9.0 supported.
+    local is_summary, summary = pcall(require, 'metrics.collectors.summary')
+    if is_summary == false or summary.rotate_age_buckets == nil then
+        return false
+    end
+
+    return true
+end
+
+--- Initialize collectors in global metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, init is not guaranteed to be idempotent.
+--  Destroy collectors only through this registry methods.
+--
+-- @function init
+--
+-- @treturn boolean Returns true.
+--
+function registry.init()
+    internal_registry[metric_name.stats] = metrics.summary(
+        metric_name.stats,
+        'CRUD router calls statistics',
+        DEFAULT_QUANTILES,
+        DEFAULT_SUMMARY_PARAMS)
+
+    internal_registry[metric_name.space_not_found] = metrics.counter(
+        metric_name.space_not_found,
+        'Spaces not found during CRUD calls')
+
+    internal_registry[metric_name.details.tuples_fetched] = metrics.counter(
+        metric_name.details.tuples_fetched,
+        'Tuples fetched from CRUD storages during select/pairs')
+
+    internal_registry[metric_name.details.tuples_lookup] = metrics.counter(
+        metric_name.details.tuples_lookup,
+        'Tuples looked up on CRUD storages while collecting response during select/pairs')
+
+    internal_registry[metric_name.details.map_reduces] = metrics.counter(
+        metric_name.details.map_reduces,
+        'Map reduces planned during CRUD select/pairs')
+
+    return true
+end
+
+--- Unregister collectors in global metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, destroy is not guaranteed to be idempotent.
+--  Destroy collectors only through this registry methods.
+--
+-- @function destroy
+--
+-- @treturn boolean Returns true.
+--
+function registry.destroy()
+    for _, c in pairs(internal_registry) do
+        metrics.registry:unregister(c)
+    end
+
+    internal_registry = {}
+    return true
+end
+
+--- Get copy of global metrics registry
+--
+--  Registries are not meant to used explicitly
+--  by users, get is not guaranteed to work without init.
+--
+-- @function get
+--
+-- @tparam string space_name
+--  (Optional) If specified, returns table with statistics
+--  of operations on table, separated by operation type and
+--  execution status. If there wasn't any requests for table,
+--  returns {}. In not specified, returns table with statistics
+--  about all existing spaces and count of calls to spaces
+--  that wasn't found.
+--
+-- @treturn table Returns copy of metrics registry.
+function registry.get(space_name)
```

Comment:
We have at least two service spaces: `_ddl_sharding_key` and `_ddl_sharding_func`. Should we exclude them from statistics?

Reply:
Well, `_ddl_sharding_key` and `_ddl_sharding_func` are called not through `crud.get()`, thus they will not be in statistics. On the other hand, if one would call `crud.insert(_ddl_sharding_key, ...)` for any reason, I think we should include this statistics.